plugins/plugin-machine/che-plugin-machine-ext-server/src/main/java/org/eclipse/che/ide/ext/machine/server/ssh/KeysInjector.java

```diff
@@ -29,97 +46,120 @@
  */
 @Singleton // must be eager
 public class KeysInjector {
+
   private static final Logger LOG = LoggerFactory.getLogger(KeysInjector.class);
 
   private final EventService eventService;
-  private final DockerConnector docker;
   private final SshManager sshManager;
-  // TODO replace with WorkspaceManager
-  //    private final CheEnvironmentEngine environmentEngine;
+  private final UserManager userManager;
+  private final WorkspaceManager workspaceManager;
+  private final ExecAgentClientFactory execAgentClientFactory;
 
   @Inject
   public KeysInjector(
-      EventService eventService, DockerConnector dockerConnector, SshManager sshManager
-      //                        CheEnvironmentEngine environmentEngine
-      ) {
+      EventService eventService,
+      SshManager sshManager,
+      UserManager userManager,
+      WorkspaceManager workspaceManager,
+      ExecAgentClientFactory execAgentClientFactory) {
     this.eventService = eventService;
-    this.docker = dockerConnector;
     this.sshManager = sshManager;
-    //        this.environmentEngine = environmentEngine;
+    this.userManager = userManager;
+    this.workspaceManager = workspaceManager;
+    this.execAgentClientFactory = execAgentClientFactory;
   }
 
   @PostConstruct
   public void start() {
     eventService.subscribe(
-        new EventSubscriber<MachineStatusEvent>() {
+        new EventSubscriber<WorkspaceStatusEvent>() {
           @Override
-          public void onEvent(MachineStatusEvent event) {
-            if (event.getEventType() == MachineStatus.RUNNING) {
-              /*final Instance machine;
+          public void onEvent(WorkspaceStatusEvent event) {
```

Comment:
This method is too long. I think it would be better to rewrite it using private methods. Then body of this method will show common logic, like:
```
@Override
public void onEvent(WorkspaceStatusEvent event) {
  final List<String> execServerURLs = getExecServerUrls(workspaceId);
  if (serverURLs.isEmpty()) {
    return; // no exec servers installed, so not possible to execute commands
  }
  
  List<String> publicKeys = getPublicKeys(workspaceId);
  if (publicKeys.isEmpty() {
    //nothing to inject
   return;
  }

  for (String execServerURL : execServerURLs) {
    doInjectPublicKeys(publicKeys);
  }
}
```
Does it make sense?